packages/react-core/src/components/Form/FormGroup.tsx

```diff
@@ -120,38 +123,48 @@ export const FormGroup: React.FunctionComponent<FormGroupProps> = ({
   );
 
   return (
-    <div
-      className={css(styles.formGroup, className)}
-      {...(role && { role })}
-      {...(isGroupOrRadioGroup && { 'aria-labelledby': `${fieldId}-legend` })}
-      {...props}
-    >
-      {label && (
+    <GenerateId>
+      {randomId => (
         <div
-          className={css(
-            styles.formGroupLabel,
-            labelInfo && styles.modifiers.info,
-            hasNoPaddingTop && styles.modifiers.noPaddingTop
-          )}
-          {...(isGroupOrRadioGroup && { id: `${fieldId}-legend` })}
+          className={css(styles.formGroup, className)}
+          {...(role && { role })}
+          {...(isGroupOrRadioGroup && { 'aria-labelledby': `${fieldId}-legend` })}
+          {...(!fieldId && { 'aria-labelledby': `${randomId}-legend` })}
```

Comment:
Rather than having both of these lines to set `aria-labelledby`, they could be condensed down to one:

```suggestion
          {...(isGroupOrRadioGroup && { 'aria-labelledby': `${fieldId || randomId}-legend` })}
```